torch/_inductor/fx_passes/freezing_patterns.py

```diff
@@ -24,15 +27,31 @@
 binary_folding_pass = PatternMatcherPass()
 
 
-def freezing_passes(gm: torch.fx.GraphModule):
+def freezing_passes(gm: torch.fx.GraphModule, aot_example_inputs):
     """
     Passes that are applied to the graph to freeze pass.
     """
 
+    from ..freezing import constant_fold
+
     lazy_init()
-    binary_folding_pass.apply(gm.graph)
-    for patterns in pass_patterns:
-        patterns.apply(gm.graph)
+    # We need a few rounds of binary folding to get rid of all the
+    # unnecessary nodes, but may need a good method to chose the rounds number.
+    # works like: conv+binary+binary.
+    binary_folding = 0
+    for _ in range(4):
+        constant_fold(gm)
+        # Make sure meta['val'] is properly set for all nodes
+        fake_tensor_prop(gm, aot_example_inputs, True)
```

Comment:
TODO: remove the need to run fake_tensor_prop on the whole model,

Reply:
added.